sdk/storage/storage-queue/RecordAndPlayback.md

```diff
@@ -0,0 +1,199 @@
+# GUIDELINES FOR RECORD AND PLAYBACK - Typescript
+
+## NOCK [for node tests]
+
+- [nock](https://www.npmjs.com/package/nock)-package is being used to test modules that perform HTTP requests.
+- To mock an existing live system, we record and playback the HTTP calls using `nock.recorder`.
+- Recording relies on intercepting real requests and responses and then persisting them for later use.
+- Nock works by overriding node’s http.request function.
+
+
+---
+
+## NISE [for browser tests]
+
+- Nock has no support for browsers. For the browser tests, we're using [nise](https://www.npmjs.com/package/nise).
+- Nise works in a way similar to Nock, intercepting HTTP requests and mocking responses.
+- Unlike Nock, Nise does not have a native record/playback feature.
+- Some Nise functions are being overwritten to enable record and playback.
+- Added new Karma plugins to access the disk and write/read recording files ([karma-json-to-file-reporter](https://www.npmjs.com/package/karma-json-to-file-reporter) to write and [karma-json-preprocessor](https://www.npmjs.com/package/karma-json-preprocessor) to read).
+
+---
+
+## Setup for record and playback
+
+**New env variable for recordings - TEST_MODE** [Supposed to be added in the `.env` file to be able to do record and playback]
+
+- If TEST_MODE = "record",
+  - Tests hit the live-service
+  - Nock/Nise are used for recording the request-responses for future use
+  - If recordings are already present, forces re-recording
+- Else If TEST_MODE = "playback",
+  - Existing recordings are used
+- Else
+  - Tests hit the live-service, we don't record the requests/responses
+
+---
+
+## Skipped tests (specific to storage-packages)
+
+- Some of the tests are being skipped because the record and playback doesn't work as expected.
```

Comment:
Improve dev convenience while skipping a test.
Currently, we are supposed to provide the file name of the recording to skip the test.
That implicitly means that the developer should have the knowledge of how a test title translates to the filename.
Instead take describe block name and the test title as inputs in the skip list.

Reply:
#4084